### PR TITLE
BXMSPROD-944 GA build-chain pull-request flow

### PR DESCRIPTION
**Pull request by Ginxo**
https://issues.redhat.com/browse/BXMSPROD-944

related PRs:
* https://github.com/kiegroup/kogito-apps/pull/535
* https://github.com/kiegroup/kogito-examples/pull/452
* https://github.com/kiegroup/kogito-runtimes/pull/887
* https://github.com/kiegroup/kogito-pipelines/pull/158

Many thanks for submitting your Pull Request :heart:! 

Please make sure that your PR meets the following requirements:

**WARNING! Please make sure you are opening your PR against `master` branch!**

- [ ] You have read the [contributors guide](https://github.com/kiegroup/kogito-runtimes#contributing-to-kogito)
- [ ] Pull Request title is properly formatted: `KOGITO-XYZ Subject`
- [ ] Pull Request title contains the target branch if not targeting master: `[0.9.x] KOGITO-XYZ Subject`
- [ ] Pull Request contains link to the JIRA issue
- [ ] Pull Request contains link to any dependent or related Pull Request
- [ ] Pull Request contains description of the issue
- [ ] Pull Request does not include fixes for issues other than the main ticket

### CODE REVIEW

File: .ci/project-dependencies.yaml [hunk 1]
```diff
@@ -0,0 +1,12 @@
+version: "2.0"
+dependencies:
+  - project: kiegroup/kogito-runtimes
+
+  - project: kiegroup/kogito-apps
+    dependencies:
+      - project: kiegroup/kogito-runtimes
```
Comment (danielezonca) [inline .ci/project-dependencies.yaml]: To be removed

File: .ci/project-dependencies.yaml [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+
+  - project: kiegroup/kogito-examples
+    dependencies:
+      - project: kiegroup/kogito-apps
```
Comment (danielezonca) [inline .ci/project-dependencies.yaml]: To be removed

File: .ci/pull-request-config.yaml [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+  - project: kiegroup/kogito-runtimes
+    build-command:
+      upstream: |
+        mvn clean install -Dmaven.wagon.httpconnectionManager.ttlSeconds=25 -Dmaven.wagon.http.retryHandler.count=3
+      current: |
+        mvn clean install -Prun-code-coverage -Dmaven.wagon.httpconnectionManager.ttlSeconds=25 -Dmaven.wagon.http.retryHandler.count=3
+        mvn validate -Psonarcloud-analysis -e -nsu -Dmaven.wagon.httpconnectionManager.ttlSeconds=25 -Dmaven.wagon.http.retryHandler.count=3
+        mvn -f kogito-runtimes-integration-tests/pom.xml clean
+    clone:
+      - kogito-runtimes-integration-tests
```
Comment (danielezonca) [inline .ci/pull-request-config.yaml]: To be removed

File: .ci/pull-request-config.yaml [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+
+  - project: kiegroup/kogito-apps
+    build-command:
+      upstream: |
+        mvn clean install -Dmaven.wagon.httpconnectionManager.ttlSeconds=25 -Dmaven.wagon.http.retryHandler.count=3
+      current: |
+        mvn clean install -Prun-code-coverage -Dmaven.wagon.httpconnectionManager.ttlSeconds=25 -Dmaven.wagon.http.retryHandler.count=3
+        mvn validate -Psonarcloud-analysis -e -nsu -Dmaven.wagon.httpconnectionManager.ttlSeconds=25 -Dmaven.wagon.http.retryHandler.count=3
```
Comment (danielezonca) [inline .ci/pull-request-config.yaml]: To be removed

File: .ci/pull-request-config.yaml [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+
+  - project: kiegroup/kogito-examples
+    build-command:
+      current: |
```
Comment (danielezonca) [inline .ci/pull-request-config.yaml]:
```suggestion
      upstream: |
        mvn clean install -Dmaven.wagon.httpconnectionManager.ttlSeconds=25 -Dmaven.wagon.http.retryHandler.count=3 -DskipTests -DskipITs
      current: |
```

File: .ci/pull-request-config.yaml (hunk 3 above)
Comment (Ginxo) [reply .ci/pull-request-config.yaml]: can you please confirm these are the lines to be removed @danielezonca 
```
      mvn -f kogito-runtimes-integration-tests/pom.xml clean
    clone:
      - kogito-runtimes-integration-tests
```

File: .ci/pull-request-config.yaml (hunk 4 above)
Comment (Ginxo) [reply .ci/pull-request-config.yaml]: done

File: .ci/project-dependencies.yaml (hunk 1 above)
Comment (Ginxo) [reply .ci/project-dependencies.yaml]: can you please confirm the dependency `kiegroup/kogito-runtimes` has to be removed?

File: .ci/project-dependencies.yaml (hunk 2 above)
Comment (Ginxo) [reply .ci/project-dependencies.yaml]: can you please confirm the dependency `project: kiegroup/kogito-apps` has to be removed? @danielezonca

File: .ci/project-dependencies.yaml (hunk 1 above)
Comment (danielezonca) [reply .ci/project-dependencies.yaml]: No, I mean that kogito-examples don't depend on `kogito-apps` so we can remove that step entirely (and simplify the FDB)

File: .ci/project-dependencies.yaml (hunk 2 above)
Comment (danielezonca) [reply .ci/project-dependencies.yaml]: Yes I confirm, it is not included in existing PR check
https://github.com/kiegroup/kogito-examples/blob/master/Jenkinsfile#L27-L31

File: .ci/pull-request-config.yaml (hunk 3 above)
Comment (danielezonca) [reply .ci/pull-request-config.yaml]: If I have understood correctly the way `upstream` work, [this configuration](https://github.com/kiegroup/kogito-examples/pull/452#discussion_r620977930) should be enough to cover also runtime compilation, thus this entire block can be removed.

File: .ci/project-dependencies.yaml (hunk 2 above)
Comment (Ginxo) [reply .ci/project-dependencies.yaml]: @danielezonca I will remove everything related to kogito-examples then...

Comment (Ginxo) [reply .ci/project-dependencies.yaml]: Sorry, it doesn't make sense (I mean what i proposed), probably we can spend some time after today's meeting to clarify the whole thing

**Review by danielezonca — APPROVED** [review]

**Pull request merged**